src/Uno.UWP/Storage/Helpers/ConcurrentEntryManager.cs

```diff
@@ -0,0 +1,59 @@
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Uno.Disposables;
+using Uno.Threading;
+
+namespace Windows.Storage.Helpers
+{
+	internal partial class ConcurrentEntryManager
+	{
+		private readonly Dictionary<string, DownloadEntry> _assetsGate = new Dictionary<string, DownloadEntry>();
+
+		public async Task<IDisposable> LockForAsset(CancellationToken ct, string updatedPath)
+		{
+			DownloadEntry GetEntry()
+			{
+				lock (_assetsGate)
+				{
+					if (!_assetsGate.TryGetValue(updatedPath, out var entry))
+					{
+						_assetsGate[updatedPath] = entry = new DownloadEntry();
+					}
+
+					entry.ReferenceCount++;
+
+					return entry;
+				}
+			}
+
+			var entry = GetEntry();
+
+			var disposable = await entry.Gate.LockAsync(ct);
```

Comment:
Codacy found an issue: [Remove this useless assignment to local variable 'disposable'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6212827)